addons/purchase_stock/static/src/product_catalog/controller/use_purchase_suggest_wizard.js

```diff
@@ -0,0 +1,64 @@
+import { onWillStart, useState, useSubEnv, useEffect } from "@odoo/owl";
+import { rpc } from "@web/core/network/rpc";
+import { debounce } from "@web/core/utils/timing";
+
+/**
+ * Hook to replicate and synchronise the state of the backend transient model
+ * purchaseSuggestWizard, and expose it to a subenv.
+ */
+export function usePurchaseSuggestWizard(poId, model) {
```

Comment:
Why using a hook instead of defining all that directly in the kanban controller ?